src/Microsoft.AspNetCore.Sockets.Client/WebSocketsTransport.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO.Pipelines;
+using System.Net.WebSockets;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+using System.Diagnostics;
+
+namespace Microsoft.AspNetCore.Sockets.Client
+{
+    public class WebSocketsTransport : ITransport
+    {
+        private ClientWebSocket _webSocket = new ClientWebSocket();
+        private IChannelConnection<Message> _application;
+        private CancellationToken _cancellationToken = new CancellationToken();
+        private readonly ILogger _logger;
+
+        public WebSocketsTransport()
+        {
+            _logger = NullLoggerFactory.Instance.CreateLogger("WebSocketsTransport");
```

Comment:
Call the other ctor with null and:

`_logger = (loggerFactory ?? NullLoggerFactory.Instance).Instance.CreateLogger("WebSocketsTransport");`

Otherwise there is NRE when you call `new WebSocketTransport(null);`